waterbox/opera/bizhawk.cpp

```diff
@@ -0,0 +1,402 @@
+#include "bizhawk.hpp"
+#include <stdlib.h>
+#include <string>
+#include <libretro.h>
+#include <lr_input.h>
+#include <opera_vdlp.h>
+#include <opera_mem.h>
+#include <opera_cdrom.h>
+#include <opera_xbus.h>
+
+struct MemoryAreas 
+{
+	uint8_t* wram;
+  uint8_t* vram;
+};
+
+struct MemorySizes
+{
+	size_t wram;
+  size_t vram;
+};
+
+#define _MAX_AUDIO_SAMPLE_COUNT 4096
+
+std::string _biosFilePath;
+std::string _gameFilePath;
+std::string _fontFilePath;
+int _port1Type;
+int _port2Type;
+controllerData_t _port1Value;
+controllerData_t _port2Value;
+MemoryAreas _memoryAreas;
+MemorySizes _memorySizes;
+uint32_t* _videoBuffer;
+size_t _videoHeight;
+size_t _videoWidth;
+size_t _videoPitch;
+int _region;
+bool _isLagFrame;
+int _nvramChanged;
+
+#define _MAX_SAMPLES 4096
+#define _CHANNEL_COUNT 2
+int16_t _audioBuffer[_MAX_SAMPLES * _CHANNEL_COUNT];
+size_t _audioSamples;
+
+extern "C"
+{  
+	   void* xbus_cdrom_plugin(int   proc_, 	void* data_);
+	   void opera_cdrom_set_callbacks(opera_cdrom_get_size_cb_t get_size_,	opera_cdrom_set_sector_cb_t set_sector_,	opera_cdrom_read_sector_cb_t read_sector_);
+	   void opera_nvram_init(void *buf, const int bufsize);
+    void opera_lr_callbacks_set_audio_sample(retro_audio_sample_t cb);
+    void opera_lr_callbacks_set_audio_sample_batch(retro_audio_sample_batch_t cb);
+    void opera_lr_callbacks_set_environment(retro_environment_t cb);
+    void opera_lr_callbacks_set_input_poll(retro_input_poll_t cb);
+    void opera_lr_callbacks_set_input_state(retro_input_state_t cb);
+    void opera_lr_callbacks_set_log_printf(retro_log_printf_t cb);
+    void opera_lr_callbacks_set_video_refresh(retro_video_refresh_t cb);
+    RETRO_API void *retro_get_memory_data(unsigned id);
+    RETRO_API size_t retro_get_memory_size(unsigned id);
+				void retro_set_controller_port_device(unsigned port_, unsigned device_);
+				void retro_get_system_av_info(struct retro_system_av_info *info_);

```

Comment:
There are mismatched spaces and tabs everywhere, could you decide on one and fix the formatting for this file?

Reply:
my IDE was playing tricks with me. Fixed now.